tests/integration/test_integ.py

```diff
@@ -0,0 +1,129 @@
+
+        # Start docker just in case it is down
+        docker_start=subprocess.Popen(
+            ["docker", "container", "start", "synthetic-file-system"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+
+        # Run mount.sh
+        docker_mount=subprocess.Popen(["docker", "container", "exec", "-d", "synthetic-file-system",
+                                         "./mount.sh"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        docker_mount_out, docker_mount_err=docker_mount.communicate()
+        docker_mount_err = docker_mount_err.decode()
+        if "failed" in docker_mount_err:
+             print("ERROR RUNNING MOUNT.SH!!!")
+             exit()
+
+        time.sleep(5)
+
+        # goes into the container and gets a list of all files under fuse_mount and writes it to a file
+        self.mounted_files=os.popen(
+            'docker container exec synthetic-file-system /bin/sh -c "cd ..; cd /fuse_mount; find . -type f" ').read()
+        f=open(self.FUSE_MOUNTED_FILES_LIST, 'w')
+        print(self.mounted_files, file=f)
+        f.close()
+
+        # Throws error if file containing mounted data is empty
+        if os.stat(self.FUSE_MOUNTED_FILES_LIST).st_size == 0:
+            print("Mounted files list at " + self.FUSE_MOUNTED_FILES_LIST, " empty!!!")
+            exit()
+
+
+
+    @pytest.mark.dependency(depends=['test_mount'])
+    def test_dir_structure(self):
+        print("/************************************* Entered test_dir_structure *************************************/")
+
+        common_dir = self.MOUNTED_FROM.split("/")
+
+        # Read all fuse files into an array
+        fuse_mounted_files=open(self.FUSE_MOUNTED_FILES_LIST, "r")
+        fuse_abs_mount_lines=fuse_mounted_files.readlines()
+        fuse_mounted_files.close()
+
+        # transforms '/home/sfsuser/Vaidehi/amos-rep/test_tree/dir0/dir144/dir158/dir160/dir164/dir165/CNV-1699976-2.jpeg' into '/dir0/dir144/dir158/dir160/dir164/dir165/CNV-1699976-2.jpeg'
+        fuse_mount_lines=[i.split(common_dir[-1], 1)[1]
+                            for i in fuse_abs_mount_lines]
+
+        # Checks if files found at mounted_from were actually mounted on the vfs
+        for root, dirs, files in os.walk(self.MOUNTED_FROM):
+            for f in files:
+                # /home/sfsuser/Vaidehi/amos-rep/test_tree/dir0/dir144/dir158/dir160/dir164/dir165/CNV-1699976-2.jpeg
+                file_path=os.path.join(root, f)
+                req_path=file_path.split(common_dir[-1])
+
+                if (req_path[1] in fuse_mount_lines) == False:
+                    self.assertRaises(FileNotFoundError)
+
+
+    # Turn off docker services
+    def tearDown(self):
+        print("/************************************* Entered tearDown *************************************/")
+
+        docker_compose_down=subprocess.Popen(["docker-compose", "down"])
```

Comment:
a built docker image should also be removed afterwards. `client.containers.stop` can also be used i think, there is lots of stuff on that referenced page

Reply:
done